src/main/kotlin/org/koitharu/kotatsu/parsers/site/HoneyMangaParser.kt

```diff
@@ -0,0 +1,210 @@
+import org.koitharu.kotatsu.parsers.util.json.getStringOrNull
+import org.koitharu.kotatsu.parsers.util.json.mapJSON
+import org.koitharu.kotatsu.parsers.util.json.mapJSONIndexed
+import org.koitharu.kotatsu.parsers.util.parseJson
+import org.koitharu.kotatsu.parsers.util.parseJsonArray
+import org.koitharu.kotatsu.parsers.util.removeSuffix
+import org.koitharu.kotatsu.parsers.util.tryParse
+import java.text.SimpleDateFormat
+import java.util.EnumSet
+import java.util.Locale
+
+/**
+ * This parser is just for parser development, it should not be used in releases
+ */
+
+private const val PAGE_SIZE = 20
+
+@MangaSourceParser("HONEY-MANGA", "Honey Manga", "uk")
+class HoneyMangaParser(context: MangaLoaderContext) : PagedMangaParser(context, MangaSource.DUMMY, PAGE_SIZE), Interceptor {
+
+	private val urlApi = "https://data.api.$domain"
+	private val mangaApi = "$urlApi/v2/manga/cursor-list"
+	private val chapterApi = "$urlApi/v2/chapter/cursor-list"
+	private val genresListApi = "$urlApi/genres-tags/genres-list"
+	private val framesApi = "$urlApi/chapter/frames"
+	private val searchApi = "https://search.api.honey-manga.com.ua/api/v1/title/search-matching?query="
+
+	private val imageStorageUrl = "https://manga-storage.fra1.digitaloceanspaces.com/public-resources"
+	override val headers
+		get() = Headers.Builder()
+			.add("User-Agent", "Mozilla/5.0 (Android 13; Mobile; rv:68.0) Gecko/68.0 Firefox/109.0")
+			.build()
+
+	override val configKeyDomain: ConfigKey.Domain
+		get() = ConfigKey.Domain("honey-manga.com.ua", null)
+
+	override val sortOrders: Set<SortOrder> = EnumSet.of(
+		SortOrder.POPULARITY,
+		SortOrder.NEWEST,
+	)
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.US)
+		val body = JSONObject()
+		body.put("mangaId", manga.url)
+		body.put("pageSize", 999999) // Hack lol (no)
+		body.put("page", 1)
+		body.put("sortOrder", "ASC")
+		val chapterRequest = webClient.httpPost(chapterApi, body).parseJson()
+		return manga.copy(
+			chapters = chapterRequest.getJSONArray("data").mapJSONIndexed() { i, jo ->
+				MangaChapter(
+					id = generateUid(jo.getString("id")),
+					name = buildString {
+						append("Том ")
+						append(jo.optString("volume", "0"))
+						append(". ")
+						append("Розділ ")
+						append(jo.optString("chapterNum", "0"))
+						if (jo.optString("title") != "Title") {
+							append(" - ")
+							append(jo.optString("title"))
+						}
+					},
+					number = i + 1,
+					url = jo.optString("chapterResourcesId"),
+					scanlator = null,
+					uploadDate = dateFormat.tryParse(jo.getString("lastUpdated")),
+					branch = null,
+					source = source
+				)
+			}
+		)
+	}
+
+	override suspend fun getListPage(
+		page: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder
+	): List<Manga> {
+		val body = JSONObject()
+		var content: JSONArray? = null
+		content = if (!query.isNullOrEmpty()) {
```

Comment:
```suggestion
		val content = if (!query.isNullOrEmpty()) {
```

Do not see any reason to make `content` nullable)

Reply:
Fixed